### PR TITLE
PARQUET-1558: Use try-with-resource in Apache Avro tests

### PR DESCRIPTION
**Pull request by Fokko**
We can use the try-with-resource pattern the implicitly close the resources such as readers and writers, provided by Avro and Parquet. This makes the code more readable since we don't have the `out.close()` everywhere.

### CODE REVIEW

**Comment by masayuki038** [comment]
+1

**Comment by Fokko** [comment]
Would it be possible to get this in? Only changes on the tests. Personally, I think the try-with-resources is more readable. Apart from that, it removes 50 lines of (boilerplate) code.

**Comment by masayuki038** [comment]
@Fokko Could you resolve the conflicts?

**Comment by Fokko** [comment]
Done! 👍

**Comment by masayuki038** [comment]
@zivanfi @gszadovszky Could you review this?

**Comment by Fokko** [comment]
Good point @zivanfi. Most of it is being done by my IDEA. I've removed the unrelated changes.

File: parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWriteOldListBehavior.java [hunk 1]
```diff
@@ -235,12 +223,13 @@ public void testAll() throws Exception {
         Schema.createArray(Schema.create(Schema.Type.INT)), integerArray);
 
     GenericFixed genericFixed = new GenericData.Fixed(
-        Schema.createFixed("fixed", null, null, 1), new byte[] { (byte) 65 });
+        Schema.createFixed("fixed", null, null, 1), new byte[]{(byte) 65});
```
Comment (zivanfi) [inline parquet-avro/src/test/java/org/apache/parquet/avro/TestReadWriteOldListBehavior.java]: (nit) Unrelated formatting-only change.

File: parquet-avro/src/test/java/org/apache/parquet/avro/TestSpecificReadWrite.java [hunk 2]
```diff
@@ -127,36 +132,38 @@ public void testFilterMatchesFinalBlockOnly() throws IOException {
...
+      assertEquals(getVwPassat().toString(), reader.read().toString());
+      assertNull(reader.read());
+    }
   }
 
   @Test
   public void testFilterOnSubAttribute() throws IOException {
     Path path = writeCarsToParquetFile(1, CompressionCodecName.UNCOMPRESSED, false);
-    
+
```
Comment (zivanfi) [inline parquet-avro/src/test/java/org/apache/parquet/avro/TestSpecificReadWrite.java]: (nit) Unrelated formatting-only change.

**Comment by Fokko** [comment]
Thanks for merging @zivanfi 

The intention was there to fix the nits could not find the time to do it
